### ISSUE TITLE
nodemon not found in PATH when running 'tilt up', but works with 'docker-compose up

### ISSUE THREAD

**Issue by theodore-q** [tilt-dev/tilt#6171]
Expected Behavior
Running tilt up should start the services as defined in the docker-compose.yml and Tiltfile, with nodemon monitoring Python application for changes and restarting the server when code changes are detected.

Current Behavior
nodemon command is not found when I run tilt up, despite being installed globally in the Docker image and working as expected when running docker-compose up. The error message I'm receiving is:

`Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "npx": executable file not found in $PATH: unknown`
Steps to Reproduce
Create the following Dockerfile in a directory named python:
Dockerfile:
```
FROM nikolaik/python-nodejs:python3.9-nodejs16
RUN npm install -g nodemon npx
```
Create the following docker-compose.yml:
yaml
```
version: '3'
services:
  python-app:
    build: ./python
    ports:
      - "50050:50050"
    volumes:
      - ./python:/app
    command: ["nodemon", "--exec", "python", "-u", "/app/app.py"]
```
Create the following Tiltfile:
```
docker_compose('docker-compose.yml')

docker_build('azureuser_python-app', './python', 
             live_update=[
                 run('python -m grpc_tools.protoc -I. --python_out=. --grpc_python_out=. ./protobuf.proto', trigger='./python/protobuf.proto'),
                 run('touch /app/app.py',  trigger='./python/**'),
                 run("ls *.py | entr sh -c 'clear; python app.py'", trigger='./python/**'),
             ])
```
Run tilt up.
Context
tilt doctor Output:
```yaml
Tilt: v0.33.1, built 2023-06-28
System: linux-amd64
---
Docker
- Host: unix:///var/run/docker.sock
- Server Version: 24.0.2
- API Version: 1.43
- Builder: 2
- Compose Version: v2.18.1
---
Kubernetes
- Env:
- Context:
- Cluster Name: Unknown
- Namespace: default
- Container Runtime: unknown
- Version: Error: invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable
- Cluster Local Registry: none
---
```
About Your Use Case
I'm trying to set up a development environment where the Python server is restarted automatically when code changes are detected. This setup works as expected when I use docker-compose up, but fails with the aforementioned error when I run tilt up. The issue is affecting my development process as it prevents me from using Tilt for its intended purpose of improving the development workflow. I suspect that the issue might be related to Docker volumes, as I have configured my Python application to use a volume in the Docker Compose file.

**Comment by nicks** [comment]
Thanks for the report! The dockerfile you provided didn't work for me. Here's the error i got:

```
 => ERROR [2/2] RUN npm install -g nodemon npx                                                                                                                                                3.5s
------
 > [2/2] RUN npm install -g nodemon npx:
2.911 npm notice 
2.911 npm notice New major version of npm available! 8.19.4 -> 9.8.0
2.911 npm notice Changelog: <https://github.com/npm/cli/releases/tag/v9.8.0>
2.911 npm notice Run `npm install -g npm@9.8.0` to update!
2.911 npm notice 
2.913 npm ERR! code EEXIST
2.913 npm ERR! path /usr/bin/npx
2.914 npm ERR! EEXIST: file already exists
2.914 npm ERR! File exists: /usr/bin/npx
2.914 npm ERR! Remove the existing file and try again, or run npm
2.914 npm ERR! with --force to overwrite files recklessly.
2.916 
2.916 npm ERR! A complete log of this run can be found in:
2.916 npm ERR!     /root/.npm/_logs/2023-07-14T16_32_31_522Z-debug-0.log
```

My guess is that your app is just broken, and the reason you're able to repro with tilt (but not docker-compose) is due to caching of an old version.

**Comment by nicks** [comment]
closing this because there's not enough information to repro
